modules/ui/org.eclipse.fx.ui.preferences/src/main/java/org/eclipse/fx/ui/preferences/page/FieldEditor.java

```diff
@@ -172,4 +195,27 @@ protected boolean displayLabel() {
 		return true;
 	}
 
+	/**
+	 * Return the observable {@link Status} property for this editor's value
+	 * 
+	 * @return
+	 */
+	public ObjectExpression<Status> statusProperty() {
+		return status.getReadOnlyProperty();
+	}
+
+	/**
+	 * Add a validation function to this field editor. The validation function
+	 * accepts a value and produces an error message as a {@link String}, or
+	 * <code>null</code> if the value is valid.
+	 * 
+	 * @param validationFunction
+	 *            A Function taking a value as a parameter, and returning a
+	 *            {@link String} error message, or <code>null</code> if the value is
+	 *            valid.
+	 */
+	public void addValidator(@NonNull Function<? super T, String> validationFunction) {
```

Comment:
How can I remove a validator? To follow efxclipse-APIs it would be named `public Subscription registerValidator(Function<? super T, Status> validationFunction)`

Reply:
> Why not make this on return a Status? This way you can also transport eg warnings?

Ok

> How can I remove a validator? To follow efxclipse-APIs it would be named public Subscription registerValidator(Function<? super T, Status> validationFunction)

Ok, I'll rename & add an unregister method

Reply:
No need for an unregister() the Subscription object is the unregister ;-)